guides/index_template.md

```diff
@@ -0,0 +1,182 @@
+# Index Template
+Index templates are a convenient way to define settings, mappings, and aliases for one or more indices when they are created. In this guide, you'll learn how to create an index template and apply it to an index.
+
+## Setup
+
+Assuming you have OpenSearch running locally on port 9200, you can create a client instance
+with the following code:
+```python
+from opensearchpy import OpenSearch
+client = OpenSearch(
+    hosts=['https://admin:admin@localhost:9200'],
```

Comment:
Separate auth, `http_auth=...`, it's not a great practice to include that in the URL.

Reply:
I have added the http_auth param and removed "admin:admin" from the hosts param. This change has also been pushed.